### PR TITLE
rawkv: fix ttl_checker for RawKV API v2

### LINKED ISSUE #15142: rawkv: `ttl-check-poll-interval` does not take effect on RawKV API v2
### What version of TiKV are you using?

master (c27b43018ca83bd4103abf627c50293b841396bd)

### Steps to reproduce

1. Set [`storage.ttl-check-poll-interval`](https://docs.pingcap.com/tidb/stable/tikv-configuration-file#ttl-check-poll-interval) to `24h`.
2. Put RawKV data with TTL of 7 days and incremental keys. Data size is generally similar everyday.

### What did you expect?

Size of storage is approximately equal for every 7 days.

### What did happened?

Size of storage keeps increasing.

### PR DESCRIPTION
**Pull request by pingyu**
### What is changed and how it works?

Issue Number: Close #15142 

What's Changed:

```commit-message
rawkv: fix ttl_checker for RawKV API v2.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- No.

### Release note 

```release-note
rawkv: fix ttl_checker for RawKV API v2.
```

### CODE REVIEW

File: tests/failpoints/cases/test_ttl.rs [hunk 1]
```diff
@@ -34,72 +37,110 @@ fn test_ttl_checker_impl<F: KvFormat>() {
...
+                }
+            }
+            kvdb.flush_cf(CF_DEFAULT, true).unwrap();
+            assert_eq!(expect_keys, 0);
+        }
+
+        gc_runner.post_gc();
+    };
+
+    let key1 = make_raw_key::<F>(b"r\0key1", Some(commit_ts));
```
Comment (ystaticy) [inline tests/failpoints/cases/test_ttl.rs:80]: How about encapsulating `make_raw_key`, make RawValue, `put_cf` and `flush_cf `  into a function?

File: tests/failpoints/cases/test_ttl.rs [hunk 2]
```diff
@@ -34,72 +37,110 @@ fn test_ttl_checker_impl<F: KvFormat>() {
...
     kvdb.flush_cf(CF_DEFAULT, true).unwrap();
-    let key2 = b"zr\0key2";
+    let key2 = make_raw_key::<F>(b"r\0key2", Some(commit_ts));
     let value2 = RawValue {
         user_value: vec![0; 10],
         expire_ts: Some(120),
         is_delete: false,
     };
-    kvdb.put_cf(CF_DEFAULT, key2, &F::encode_raw_value_owned(value2))
+    kvdb.put_cf(CF_DEFAULT, &key2, &F::encode_raw_value_owned(value2))
```
Comment (ystaticy) [inline tests/failpoints/cases/test_ttl.rs:95]: Is there need do `flush_cf` after put `key2`?

Comment (pingyu) [reply tests/failpoints/cases/test_ttl.rs:95]: I think yes. `flush_cf` flush memtable to sst, and `compact` operates on sst files only.

File: tests/failpoints/cases/test_ttl.rs (hunk 1 above)
Comment (pingyu) [reply tests/failpoints/cases/test_ttl.rs]: I refactor to a loop to generate test data, for eliminating duplicated codes. PTAL~
https://github.com/tikv/tikv/pull/15143/commits/bfa18d88923ff4ab822003abe16ca618c73a4c64

**Review by MyonKeminta — APPROVED** [review]

**Pull request merged**